### DISCUSSION TITLE
Version 4 io.sockets.connected returns undefined

### DISCUSSION THREAD

**Discussion by taylordwright44** [socketio/socket.io#3993] [Q&A] [answered]
I had an app that used io.sockets.connected[target.socketId].leave(room) to kick someone out of a room via their socket.id. I upgraded to 4.1.0 recently on a new app and using the same code it doesn't work, and when I tried console.log(io.sockets.connected) it says undefined. 

What's the proper way to kick someone out of a room by their socket ID now? I can't seem to find it anywhere in the documentation. I've only found how to make the requested user leave by their socket.

**Comment by darrachequesne** [comment, marked as answer]
The `connected` attribute was indeed removed in `socket.io@3.0.0`.

See here: https://socket.io/docs/v4/migrating-from-2-x-to-3-0/#Namespace-connected-is-renamed-to-Namespace-sockets-and-is-now-a-Map

You can either:

- use the `sockets` attribute instead

```js
if (io.sockets.sockets.has(target.socketId) {
  io.sockets.sockets.get(target.socketId).leave(room);
}
```

Reference: https://socket.io/docs/v4/server-api/#namespace-sockets

- or use the `socketsLeave` method (preferred way, as it works in a multi server setup)

```js
io.in(target.socketId).socketsLeave(room);
```

Reference: https://socket.io/docs/v4/server-api/#server-socketsLeave-rooms

**Reply by taylordwright44** [reply]
Thanks mate. I couldn't seem to find any other answers. This was my first time upgrading since 2 came out.

**Reply by darrachequesne** [reply]
You're welcome! Please ping me if you have any problem while upgrading.
